### PR TITLE
Fixing #72

### PR DESCRIPTION
**Pull request by ollielo**
@junghans 

I finally got both Fedora and Ubuntu docker build working. There are still some work to be done for TPL, Legion and Coverage but it should be trivial for you to fill the gap.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,62 +1,34 @@
...
-          -DENABLE_UNIT_TESTS=ON ${COVERAGE:+-DENABLE_COVERAGE_BUILD=ON} .. && 
-    make -j2 && make test && make install DESTDIR="${HOME}" && cd ..
+# The docker package on Travis is too old to support the --build-arg option,
+# we have to use "sed" to modify the fedora to propogate the environment
+# variables set in the matrix: section.
+ - sed -i 's/\$MPI/'"${MPI}/" docker/*
+ - sed -i 's/\$RUNTIME/'"${RUNTIME}/" docker/*
+ - sed -i 's/\$WERROR/'"${WERROR}/" docker/*
+ - docker build -t flecsi_fedora -f docker/fedora docker/
+ - docker build -t flecsi_ubuntu -f docker/ubuntu docker/
```
Comment (junghans) [inline .travis.yml]: Can you make the fedora and the ubuntu into two sets of test to halfen the runtime.

**Comment by junghans** [comment]
Can you rebase your commits into 2 or 3 more useful ones?

**Comment by ollielo** [comment]
Sure, but I need some help using git rebase.

**Comment by junghans** [comment]
Forget it about that, github support squash merge now:
https://github.com/blog/2141-squash-your-commits

**Comment by junghans** [comment]
ttp://mjg59.dreamwidth.org/42759.html

**Comment by junghans** [comment]
I made some progess, but `--build-arg` don't seem to work!

**Comment by junghans** [comment]
Switching to `language: generic` already triggers the installation of docker v1.9.0, which has `--build-arg`.

**Comment by ollielo** [comment]
@junghans 

I managed to get all builds that do not have Legion working. This should be another good point for you to merge to the master.

**Comment by junghans** [comment]
Yeah, the build break hints towards another issue, we don't want to build `master` inside docker, but `${TRAVIS_COMMIT}` plus let's wait for #81.

**Comment by junghans** [comment]
I think this is ready to merge now, but I really don't like the fact that it takes travis much longer to find a free slot for testing.

**Comment by ollielo** [comment]
Is this a instance/node allocation/scheduling issue or a compile/build time
issue?

**Comment by junghans** [comment]
It is because docker runs on the legacy infrastructure, which has less machines these days.

**Comment by junghans** [comment]
@ddaniel : Your call!

**Comment by ddaniel** [comment]
@ollielo @junghans 

Is the variable MININAL a typo for MINIMAL?  If not what does it mean?

**Comment by ollielo** [comment]
Thanks.

Ollie

**Pull request merged**